### DISCUSSION TITLE
Issues with adding new webhooks

### DISCUSSION THREAD

**Discussion by bgibers** [kinngh/shopify-node-express-mongodb-app#79] [Q&A] [answered]
I've tried adding several new webhooks but they seem to get completely ignored. Here is my current registrar..and the scope is set to read_orders in the .env. I've tried blowing away the proj and rebuilding. But the new webhook is never recognized. 

```const webhookRegistrar = async () => {
  shopify.webhooks.addHandlers({
    APP_UNINSTALLED: {
      deliveryMethod: DeliveryMethod.Http,
      callbackUrl: "/webhooks/app_uninstalled",
      callback: appUninstallHandler,
    },
    ORDERS_PAID: {
      deliveryMethod: DeliveryMethod.Http,
      callbackUrl: "/webhooks/orders_paid",
      callback: orderPaidHandler,
    }
  });```

**Comment by kinngh** [comment]
Just tested it out - it seems to be working as expected. I'm assuming you haven't touched anything else like how the webhooks are registered and related stuff. I updated my `webhookRegistrar()` as:

```javascript
const webhookRegistrar = async () => {
  shopify.webhooks.addHandlers({
    APP_UNINSTALLED: {
      deliveryMethod: DeliveryMethod.Http,
      callbackUrl: "/webhooks/app_uninstalled",
      callback: appUninstallHandler,
    },
    ORDERS_PAID: {
      deliveryMethod: DeliveryMethod.Http,
      callbackUrl: "/webhooks/app_uninstalled",
      callback: appUninstallHandler,
    },
  });
};
```
I've used the same function because we are just concerned about testing if webhook is registered, and it is. When you add a new webhook make sure you're doing two things:
- Updating scopes (which you have, or it'll throw an error)
- Go through the auth flow again. If you want to force reauth, just delete the sessions, open the app and go to any page that's making a GraphQL / request to the express server and it'll reauth.

If you're still running into issues, please ping me at [@kinngh](https://twitter.com/kinngh) and we can hop on a call

![Screenshot 2023-05-18 at 2 32 52 PM](https://github.com/kinngh/shopify-node-express-mongodb-app/assets/773555/73719708-f852-45fa-8407-a9c67c59f0f9)

**Reply by bgibers** [reply]
Yeah its still not working for some reason. Pinging you now.

**Reply by bgibers** [reply]
All steps above didn't resolve the issue. Just posting that here for anyone else that may run into this. I removed the app_uninstalled webhook, but left the new one. It deletes that but still doesn't publish the new webhook. I've tried other topics as well.

**Reply by kinngh** [reply, marked as answer]
Resolved: Didn't have protected customer data access.

**Comment by Anshu-Shar** [comment]
Hello @kinngh 
I am working with Shopify remix app with SQLite database. I have configured the products_update webhook this is my webhooks.JSX file code.
![image](https://github.com/kinngh/shopify-node-express-mongodb-app/assets/81371751/76e201b5-7a2a-4eb3-90de-db2bad89bbeb)
another is shopify.server.js file code-
![image](https://github.com/kinngh/shopify-node-express-mongodb-app/assets/81371751/02efbe8e-89f8-4650-8e1d-db74353d223c)
my env file is-
![image](https://github.com/kinngh/shopify-node-express-mongodb-app/assets/81371751/1bebbf52-83b1-448b-8fd5-a0992bf62195)
The shopify.app.toml file code is-
![image](https://github.com/kinngh/shopify-node-express-mongodb-app/assets/81371751/b6cf385a-5e1e-404a-8abd-74528cbdf198)
when I use the-npm run shopify webhook trigger command I receive the response with the correct topic but the session is undefined and the shop name is also incorrect.

![image](https://github.com/kinngh/shopify-node-express-mongodb-app/assets/81371751/04eebbdf-1e01-40e2-9274-700ca33b1ea9)
Please help me out what is the problem in my code. am I missing something? please guide me.

**Reply by kinngh** [reply]
Invalid.

This repositories issues and discussions only deal with issues with the repository and not general Shopify help. Please refer to docs at https://shopify.dev or join Shopify's Dev Discord^

**Reply by Anshu-Shar** [reply]
Ok Thanks
